Lib/test/test_array.py

```diff
@@ -1014,6 +1014,22 @@ def test_pop(self):
             array.array(self.typecode, self.example[3:]+self.example[:-1])
         )
 
+    def test_clear(self):
+        a = array.array(self.typecode, self.example)
+        self.assertRaises(TypeError, a.clear, 42)
+        a.clear()
+        self.assertEqual(a, array.array(self.typecode))
+
+        a = array.array(self.typecode)
+        a.clear()
+        self.assertEqual(a, array.array(self.typecode))
+
+        a = array.array(self.typecode, self.example)
+        a.clear()
+        a.append(self.example[2])
+        a.append(self.example[3])
+        self.assertEqual(a, array.array(self.typecode, self.example[2:4]))
+
```

Comment:
Add a test that an array with exported buffers cannot be cleared, something like this (untested):

```
with memoryview(a):
    with self.assertRaises(BufferError):
        a.clear()
```

Reply:
done